include/zephyr/drivers/sensor_types.h

```diff
@@ -0,0 +1,264 @@
+/**
+ * @file drivers/sensor_types.h
+ *
+ * @brief Public APIs for the sensor types.
+ *
+ * The SENSOR_TYPE_* defines are the sensor types supported. Additional types can be added at
+ * SENSOR_TYPE_VENDOR_START.
+ */
+
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef INCLUDE_DRIVERS_SENSOR_TYPES_H_
+#define INCLUDE_DRIVERS_SENSOR_TYPES_H_
+
+#include <zephyr/math/util.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Start value for all of the vendor-defined private sensors.
+ */
+#define SENSOR_TYPE_VENDOR_START UINT8_C(192)
```

Comment:
Does the API require that this is absolutely guaranteed to be a `uint8_t`? If not, I think you should make a sensor type enum. It would make this a little more terse, and it might be more obvious that there aren't any duplicate values.

Nit: Either way, stylistically, the 4-6 lines of comments and whitespace per numeric constant are a bit tedious, and they make it hard to just scan through the whole list of types and see what is there. Is there a style-compliant way of doing this that would just have a one-liner comment per type with a con